src/coreclr/jit/morph.cpp

```diff
@@ -14342,36 +14369,27 @@ void Compiler::fgMorphTreeDone(GenTree* tree,
         return;
     }
 
-    if ((oldTree != nullptr) && (oldTree != tree))
-    {
-        /* Ensure that we have morphed this node */
-        assert((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) && "ERROR: Did not morph this node!");
-
 #ifdef DEBUG
-        TransferTestDataToNode(oldTree, tree);
-#endif
-    }
-    else
+    if ((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) == 0)
     {
-        // Ensure that we haven't morphed this node already
-        assert(((tree->gtDebugFlags & GTF_DEBUG_NODE_MORPHED) == 0) && "ERROR: Already morphed this node!");
-    }
-
-    if (tree->OperIsConst())
-    {
-        goto DONE;
+        assert("ERROR: Did not morph this node!");
```

Comment:
```suggestion
        assert(!"ERROR: Did not morph this node!");
```
Is it intentional that we no longer check for double morphing?

Reply:
Ah, thanks. That checking depended on old tree so I'd have to pass something like that down.

Seems like an odd contract though.